src/currencyMapper.ts

```diff
@@ -0,0 +1,26 @@
+import {newMonetaryAmount} from "@interlay/interbtc-api";
+import { Bitcoin, Currency, Interlay, Kintsugi, Kusama, MonetaryAmount, Polkadot, UnitList } from "@interlay/monetary-js";
+import { BigSource } from "big.js";
+
+export type Factory<U extends UnitList> = (amount: BigSource, base?: boolean) => MonetaryAmount<Currency<U>, U>
+
+export function currencyNameToCurrency<U extends UnitList>(name: string): Currency<U> {
+    let currency: any;
+    switch (name) {
+        case "Bitcoin": currency = Bitcoin;
+            break;
+        case "Polkadot": currency = Polkadot;
+            break;
+        case "Kusama": currency = Kusama;
+            break;
+        case "Interlay": currency = Interlay;
+            break;
+        case "Kintsugi": currency = Kintsugi;
+            break;
+    }
+    return currency;
+}
+
+export function currencyFactory<U extends UnitList>(currency: any): Factory<U> {
```

Comment:
Would it work if `Currency<U>` was used instead of `any`? I'm hoping it would allow us to avoid the type assertions